src/idpf.rs

```diff
@@ -308,11 +308,11 @@ where
     out
 }
 
-fn gen_with_rand_source<VI, VL, M: IntoIterator<Item = VI>, P, const SEED_SIZE: usize>(
+fn gen_with_random<VI, VL, M: IntoIterator<Item = VI>, P, const SEED_SIZE: usize>(
     input: &IdpfInput,
     inner_values: M,
     leaf_value: VL,
-    rand_source: RandSource,
+    random: &[u8],
```

Comment:
Feel free to ignore this idea:
```suggestion
    random_1: [u8; SEED_SIZE],
    random_2: [u8; SEED_SIZE],
```

Then I think you could get rid of the check on the length as well as a couple awkward `.try_into().unwrap()`s below.